.github/workflows/cmake_ctest.yml

```
@@ -0,0 +1,29 @@
+name: Cmake build and tests
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    types:
+      - synchronize
+      - opened
+  workflow_dispatch:
+
+jobs:
+  build_test:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: install extra dependencies
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y libtiff-dev
+    - name: build
+      run: |
+        cmake .
+        make -j4
+    - name: test
+      run: |
+        ctest -j4
```

```suggestion
        ctest --parallel
```

For consistency, really.